server/protocols/webadmin/webadmin-mailqueue/src/test/java/org/apache/james/webadmin/routes/MailQueueRoutesTest.java

```diff
@@ -636,7 +638,7 @@ public void deleteMailsShouldDeleteMailsWhenTheyAreMatching() throws Exception {
         .when()
             .delete(FIRST_QUEUE + "/mails")
         .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
+            .statusCode(HttpStatus.CREATED_201);
 
         MailAddress deletedRecipientMailAddress = new MailAddress(recipient);
```

Comment:
We need to `await` task completion before checking results (overwize we are exposed to data-races and thus unstable tests). Check MailRepositoriesRoutesTest::clearTaskShouldRemoveAllTheMailsFromTheMailRepository to get an example...

We need a test about task details. See MailRepositoriesRoutesTest::clearTaskShouldHaveDetails to get an hint about what to do :-).